pkgs/development/tools/analysis/ida-free/update.sh

```diff
@@ -0,0 +1,151 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p bash coreutils curl jq
+
+urls=()
+urls+=(https://out7.hex-rays.com/files/idafree82_windows.exe)
+urls+=(https://out7.hex-rays.com/files/idafree82_linux.run)
+urls+=(https://out7.hex-rays.com/files/idafree82_mac.app.zip)
+urls+=(https://out7.hex-rays.com/files/arm_idafree82_mac.app.zip)
+
+hashPrefix=sha256:
+
+cd "$(dirname "$0")"
+
+source_json_file=src.json
+
+old_source_json='{}'
+if [ -e $source_json_file ]; then
+    # parse old source info
+    old_source_json="$(< $source_json_file)"
+fi
+
+new_source_json="$old_source_json"
+
+# parse hashes
+if true; then
+html=$(curl -s https://hex-rays.com/ida-free/)
+else
+html=$(cat <<'EOF'
+test
+https://hex-rays.com/ida-free/
+b77a56913d669240b53f1cfd2deec20c3f4b00cbf286bedae46172e11f06431a  arm_idafree82_mac.app.zip
+a5cc9bf0cdfd14956c594010897cb5ee73ad55484f90fb04db862a84af6b2b9f  idafree82_linux.run
+60cd970c6aa01e14f488ef28c6017eca28bf447509f40176a1a982bb8e90f12a  idafree82_mac.app.zip
+2289a03c9f2f2c2fa3de86453139302108a32abf3b98d27267b2c9bfa48056cf  idafree82_windows.exe
+EOF
+)
+fi
+
+hash_regex='[0-9a-f]{64}' # sha256base16
+
+hashes_raw=$(echo "$html" | grep -E "^$hash_regex  [a-z0-9_.]*idafree[a-z0-9_.]+\.(run|zip|exe)$")
+
+echo "hashes_raw:"
+echo "$hashes_raw"
+
+declare -A hashes
+while read line; do
+    sha256base16=$(echo "$line" | sed -E "s/^($hash_regex)  (.+)$/\1/")
+    file=$(echo "$line" | sed -E "s/^($hash_regex)  (.+)$/\2/")
+    hashes[$file]=$sha256base16
+done <<< "$hashes_raw"
+
+for url in "${urls[@]}"; do
+    file=$(basename "$url")
+
+    # parse platform from $file
+    # arm_idafree82_mac.app.zip
+    if echo "$file" | grep -qE '^arm_idafree[0-9]+_mac.app.zip$'; then
+        platform=armv7a-darwin
```

Comment:
is this synonymous with `aarch64-darwin`?

Reply:
my bad, this should be aarch64-darwin (64bit) not armv7a-darwin (32bit)